pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/Worker.java

```diff
@@ -38,6 +50,13 @@
     private final WorkerService workerService;
     private WorkerServer server;
 
+    private ZooKeeperClientFactory zkClientFactory = null;
```

Comment:
Curious as to why the worker server needs access to zookeeper? Can we just pass the Authorization provider to the worker and decouple the implementation?

Reply:
@cckellogg this is worker and not the worker service.  This code is run when the worker and workerservice is running separately than the broker.  Look at the code for WorkerService,  I do pass in both the authentication provider and authorization provider to it. If we are running the function worker with the broker, than the PulsarService will pass those in and if not, this class, Worker will pass those in

Reply:
I think @cckellogg is asking why worker needs zk in the first place? Shouldn't all information about auth service be encapsulated in the impl(in this case pulsarauthprovider)